src/keys/YkChallengeResponseKey.cpp

```diff
@@ -43,35 +39,17 @@ YkChallengeResponseKey::YkChallengeResponseKey(YubiKeySlot keySlot)
 {
 }
 
-YkChallengeResponseKey::~YkChallengeResponseKey()
-{
-    if (m_key) {
-        gcry_free(m_key);
-        m_keySize = 0;
-        m_key = nullptr;
-    }
-}
-
-QByteArray YkChallengeResponseKey::rawKey() const
-{
-    return QByteArray::fromRawData(m_key, static_cast<int>(m_keySize));
-}
-
 bool YkChallengeResponseKey::challenge(const QByteArray& challenge)
 {
     m_error.clear();
-    QByteArray key;
+    QByteArray response;
     auto result =
-        AsyncTask::runAndWaitForFuture([&] { return YubiKey::instance()->challenge(m_keySlot, challenge, key); });
+        AsyncTask::runAndWaitForFuture([&] { return YubiKey::instance()->challenge(m_keySlot, challenge, response); });
 
     if (result == YubiKey::SUCCESS) {
-        if (m_key) {
-            gcry_free(m_key);
-        }
-        m_keySize = static_cast<std::size_t>(key.size());
-        m_key = static_cast<char*>(gcry_malloc_secure(m_keySize));
-        std::memcpy(m_key, key.data(), m_keySize);
-        sodium_memzero(key.data(), static_cast<std::size_t>(key.capacity()));
+        m_key.resize(response.size());
+        Botan::copy_mem(m_key.data(), response.constData(), response.size());
+        Botan::secure_scrub_memory(response.data(), response.capacity());
```

Comment:
`response` should also be a `secure_vector`.

Reply:
Excellent, just used m_key directly!